### PR TITLE
[AppService] az appservice domain create: Fix get correct domain agreements

### LINKED ISSUE #18651: WebApp:az appservice domain create: incorrect domain agreements
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
Incorrect domain agreements used in `az appservice domain create`. This causes domain preflight to fail on the API side

**To Reproduce**
Run `az appservice domain create` without privacy flag. The domain agreement returns privacy & transfer agreements

**Expected behavior**
privacy agreeement should only be shown when privacy is specified, and transfer agreement shouldn't be shown (since we don't support transfer)

### PR DESCRIPTION
**Pull request by calvinsID**
**Description**
Get correct domain agreements for user to acknowledge, in `az appservice domain create` command

Backend is introducing a validation API for domain creates, and domains created through CLI are passing in different agreements than what was expected causing the validation API to fail

Fixes #18651 

**Testing Guide**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

**Review by panchagnula — CHANGES_REQUESTED** [review]
are there any tests we can add here to identify these issues?

**Comment by calvinsID** [comment]
> are there any tests we can add here to identify these issues?

I don't think we can add a test for `domain create` since we aren't hooked up to GD test endpoints

**Comment by panchagnula** [comment]
> > are there any tests we can add here to identify these issues?
> 
> I don't think we can add a test for `domain create` since we aren't hooked up to GD test endpoints

The question mainly was can we create a test to validate the values returned from the command are correct (like defaults) if possible? If not this LGTM

**Comment by calvinsID** [comment]
> tion mainly was can we create a test to validate the values returned from the command are correct (li

Ah right there is a --dryrun option, let me add tests

**Review by panchagnula — APPROVED** [review]
Ship It

File: src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py [hunk 1]
```diff
@@ -50,6 +50,16 @@ def create_domain(cmd, resource_group_name, hostname, contact_info, privacy=True
     web_client = web_client_factory(cmd.cli_ctx)
     hostname_availability = web_client.domains.check_availability(NameIdentifier(name=hostname))
 
+    if not hostname_availability.available:
+        raise CLIError("Custom domain name '{}' is not available. Please try again "
+                       "with a new hostname.".format(hostname))
+
+    tld = '.'.join(hostname.split('.')[1:])
+    from azure.mgmt.web.models import TopLevelDomainAgreementOption
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py:58]: please use get_models() as 3.  in https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#multi-api-aware-modules

File: src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py [hunk 2]
```diff
@@ -50,6 +50,16 @@ def create_domain(cmd, resource_group_name, hostname, contact_info, privacy=True
     web_client = web_client_factory(cmd.cli_ctx)
     hostname_availability = web_client.domains.check_availability(NameIdentifier(name=hostname))
 
+    if not hostname_availability.available:
+        raise CLIError("Custom domain name '{}' is not available. Please try again "
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py:54]: please use error type in azclierror, e.g.
https://github.com/Azure/azure-cli/blob/3eea7fd773722b7a3c40b8115f29a3bad39b70bc/src/azure-cli-core/azure/cli/core/azclierror.py#L152-L154

**Comment by panchagnula** [comment]
@Juliehzl this PR is needed for S189 as well - can you please sign-off & merge?

**Pull request merged**